### PR TITLE
Allow to define permissions for each datalayer instead of for the whole map

### PR DESCRIPTION
**Pull request by yohanboniface**
Work in progress!

cf #584 

Todo list:

- [x] Create DataLayer.edit_status and expose in the API
- [x] adapt `can_edit` logic
- [x] allow to edit `edit_status` in the frontend
- [x] take it into account to allow or not editing feature
- [x] Expose only editable layers for new feature / editing feature's layer
- [x] make it work with anonymous maps
- [x] populate DataLayer.edit_status values from Map.edit_status
- [x] hide frontend map forms to users that do not have right to save
- [x] only show buttons create/delete datalayers to users with "map" access level
- [x] remove Map.edit_status from frontend/API
- [ ] ~delete Map.edit_status (certainly not in this PR just in case)~
- [x] backend tests
- [x] adapt frontend tests
- [x] integration tests (playwright)
- [x] decide if we keep both OWNER and EDITORS modes at datalayer level (this distinction does not exist anymore at map level)
- [x] naming (allowEdit / canEdit / allowMapEdit, hasDirty / isDIrty)
- [x] check delete map button for editors
- [x] check context menu
- [x] set default value for `Datalayer.options.editMode`
- [ ] ~add unittests for `editMode` value returned by back-office~ (it's json dumped in a `<script>` in a HTML page, so it's better to use integration tests, which test the behaviour instead of the key itself, and are already done)

### CODE REVIEW

File: umap/forms.py [hunk 1]
```diff
@@ -8,8 +8,12 @@
 
 from .models import Map, DataLayer
 
-DEFAULT_LATITUDE = settings.LEAFLET_LATITUDE if hasattr(settings, "LEAFLET_LATITUDE") else 51
-DEFAULT_LONGITUDE = settings.LEAFLET_LONGITUDE if hasattr(settings, "LEAFLET_LONGITUDE") else 2
+DEFAULT_LATITUDE = (
```
Comment (yohanboniface) [inline umap/forms.py:11]: Black noise :(

Comment (davidbgk) [reply umap/forms.py:11]:
```suggestion
DEFAULT_LATITUDE = getattr(settings, "LEAFLET_LATITUDE", 51)
```

Even shorter 😎

File: umap/static/umap/js/umap.js [hunk 2]
```diff
@@ -192,16 +192,15 @@ L.U.Map.include({
       this
     )
 
-    let isDirty = false // global status
+    let isDirty = false // self status
     try {
       Object.defineProperty(this, 'isDirty', {
         get: function () {
-          return isDirty || this.dirty_datalayers.length
+          return isDirty
```
Comment (yohanboniface) [inline umap/static/umap/js/umap.js:199]: Not sure why this check on `dirty_datalayers` was here before. As I can see, the `isDirty` status should always be set when a datalayer is made dirty…

File: umap/static/umap/js/umap.js [hunk 3]
```diff
@@ -192,16 +192,15 @@ L.U.Map.include({
...
         },
         set: function (status) {
-          if (!isDirty && status) self.fire('isdirty')
           isDirty = status
-          self.checkDirty()
+          this.checkDirty()
```
Comment (yohanboniface) [inline umap/static/umap/js/umap.js:203]: Not sure this change is harmless. It works in our browser, but may it break on some older versions ?

**Pull request merged**